opentelemetry-exporter-cloud-trace/tests/test_cloud_trace_exporter.py

```diff
@@ -182,28 +185,102 @@ def test_extract_status(self):
             Status(details=None, code=2, message="error_desc"),
         )
 
-    def test_extract_attributes(self):
+    def test_extract_empty_attributes(self):
         self.assertEqual(
             _extract_attributes({}, 4), ProtoSpan.Attributes(attribute_map={})
         )
+
+    def test_extract_variety_of_attributes(self):
         self.assertEqual(
             _extract_attributes(self.attributes_variety_pack, 4),
             self.extracted_attributes_variety_pack,
         )
-        # Test ignoring attributes with illegal value type
+
+    def test_ignore_invalid_attributes(self):
         self.assertEqual(
-            _extract_attributes({"illegal_attribute_value": dict()}, 4),
-            ProtoSpan.Attributes(attribute_map={}, dropped_attributes_count=1),
+            _extract_attributes(
+                {"illegal_attribute_value": dict(), "legal_attribute": 3}, 4
+            ),
+            ProtoSpan.Attributes(
+                attribute_map={"legal_attribute": AttributeValue(int_value=3)},
+                dropped_attributes_count=1,
+            ),
         )
 
+    def test_too_many_attributes(self):
         too_many_attrs = {}
         for attr_key in range(5):
             too_many_attrs[str(attr_key)] = 0
         proto_attrs = _extract_attributes(too_many_attrs, 4)
```

Comment:
To make this function a little more self-documenting you could write it as:

```
def test_too_many_attributes(self):
    attr_limit = 4
    too_many_attrs = {}
    for attr_key in range(attr_limit + 1):
        too_many_attrs[str(attr_key)] = 0
    proto_attrs = _extract_attributes(too_many_attrs, attr_limit)
    self.assertEqual(proto_attrs.dropped_attributes_count, 1)
```

Alternatively, using `num_attrs_limit=4` instead of just `4` as mentioned in another comment would be enough to improve the readability of this test.